src/Appwrite/Platform/Workers/Databases.php

```diff
@@ -515,19 +515,22 @@ protected function deleteCollection(Document $database, Document $collection, Do
         $databaseId = $database->getId();
         $databaseInternalId = $database->getInternalId();
 
-        $relationships = \array_filter(
-            $collection->getAttribute('attributes'),
-            fn ($attribute) => $attribute['type'] === Database::VAR_RELATIONSHIP
-        );
+        /**
+         * Related collections relating to current collection
+         */
+        $attributes = $dbForProject->find('attributes', [
+            Query::equal('databaseInternalId', [$databaseInternalId]),
+            Query::equal('type', [Database::VAR_RELATIONSHIP]),
+            Query::notEqual('collectionInternalId', $collectionInternalId),
+            Query::contains('options', ['"relatedCollection":"'. $collectionId .'"']),
+            Query::limit(PHP_INT_MAX)
```

Comment:
We usually paginate don't we?

Reply:
Using `deleteByGroup` will resolve this 👌

Reply:
Done!